### PR TITLE
Added CurlHeader tester to test curl output

### PR DESCRIPTION
**Pull request by rpm4118**
This tester compares curl output headers and values in an order-insensitive fashion. The detailed usage is explained in readme.md in autest-site directory.

### CODE REVIEW

**Comment by randall** [comment]
[approve ci]

**Comment by dragon512** [comment]
@rpm4118 Can you add a modified autest that benefits from this new tester object

File: tests/gold_tests/autest-site/curl_header.test.ext [hunk 1]
```diff
@@ -0,0 +1,198 @@
+from autest.testers import Tester, tester
```
Comment (dragon512) [inline tests/gold_tests/autest-site/curl_header.test.ext]: Needs licence headers

**Comment by dragon512** [comment]
[approve ci]

File: tests/gold_tests/headers/cache_and_req_body_curlheader.test.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+'''
```
Comment (dragon512) [inline tests/gold_tests/headers/cache_and_req_body_curlheader.test.py]: Can you modify the original test as we don't want to test the same thing twice... we just want the original test to be more reliable.

**Comment by dragon512** [comment]
[approve ci]

**Pull request merged**